samples/nrf9160/modem_shell/src/utils/str_utils.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <assert.h>
+#include <strings.h>
+#include <stdio.h>
+#include <zephyr.h>
+
+#include "str_utils.h"
+
+static inline uint8_t char2int(char input)
+{
+	if (input >= '0' && input <= '9') {
+		return input - '0';
+	}
+	if (input >= 'A' && input <= 'F') {
+		return input - 'A' + 10;
+	}
+	if (input >= 'a' && input <= 'f') {
+		return input - 'a' + 10;
+	}
+
+	return 0;
+}
+
+int str_hex_to_bytes(char *str, uint32_t str_length, uint8_t *buf,
```

Comment:
There's `hex2bin()` in `sys/utils.h`, Unless there's something I miss, you could use it instead.

Reply:
The only difference is that our version removes spaces. I've `str_hex_to_bytes()` to use `hex2bin()`.
`str_utils.c` becomes very small but I didn't remove it at least yet. Just tell me if you have an opinion on that.

Reply:
@rlubos Please resolve this if OK.